core/src/Kokkos_CopyViews.hpp

```diff
@@ -3278,26 +3278,25 @@ impl_realloc(Kokkos::View<T, P...>& v, const size_t n0, const size_t n1,
   const bool sizeMismatch = Impl::size_mismatch(v, v.rank_dynamic, new_extents);
 
   if (sizeMismatch) {
-    v = view_type();  // Deallocate first, if the only view to allocation
-    v = view_type(arg_prop, n0, n1, n2, n3, n4, n5, n6, n7);
+    auto const& label = v.label();  // Save the label
+    v = view_type();  // Best effort to deallocate in case no other view refers
+                      // to the shared allocation
+    using alloc_prop = Impl::ViewCtorProp<ViewCtorArgs..., std::string>;
+    alloc_prop arg_prop_copy(arg_prop);
+    static_cast<Kokkos::Impl::ViewCtorProp<void, std::string>&>(arg_prop_copy)
+        .value = label;
+    v          = view_type(arg_prop_copy, n0, n1, n2, n3, n4, n5, n6, n7);
```

Comment:
Why not just
```suggestion
    using alloc_prop = Impl::ViewCtorProp<ViewCtorArgs..., std::string>;
    alloc_prop arg_prop_copy(arg_prop);
    static_cast<Kokkos::Impl::ViewCtorProp<void, std::string>&>(arg_prop_copy)
        .value = v.label();  // Save the label
    v = view_type();  // Best effort to deallocate in case no other view refers
                      // to the shared allocation
    v          = view_type(arg_prop_copy, n0, n1, n2, n3, n4, n5, n6, n7);
```

Reply:
Sure